File: api-docs/packages/open-meteo-lk-docs/python/smoke.py
"""Quick client smoke — polite delay on."""

from __future__ import annotations

import json

from open_meteo_lk_docs import OpenMeteoLkDocsClient


def json_preview(data: object, limit: int = 200) -> str:
    try:
        return json.dumps(data, ensure_ascii=False)[:limit]
    except TypeError:
        return str(data)[:limit]


with OpenMeteoLkDocsClient(default_delay_seconds=1.0) as client:
    print("smoke", client.slug, "->", "forecast_colombo")
    data = client.forecast_colombo()
    preview = data[:200] if isinstance(data, str) else json_preview(data)
    print("ok", preview)

File: api-docs/packages/open-meteo-lk-docs/javascript/examples/smoke.mjs
import { OpenMeteoLkDocsClient } from "../client.mjs";

const client = new OpenMeteoLkDocsClient({ defaultDelayMs: 1000 });
console.log("smoke", OpenMeteoLkDocsClient.slug, "->", "forecastColombo");
const data = await client.forecastColombo();
const preview = typeof data === "string" ? data.slice(0, 200) : JSON.stringify(data)?.slice(0, 200);
console.log("ok", preview);
